src/core/CoreModuleDirector.ts

```diff
@@ -218,9 +257,9 @@ export class CoreModuleDirector {
     logMethodCall('CoreModuleDirector.getPushSubscriptionModelByCurrentToken');
     const pushToken = await MainHelper.getCurrentPushToken();
     if (pushToken) {
-      return this.getSubscriptionOfTypeWithToken(
-        ModelName.PushSubscriptions,
-        pushToken,
+      const pushSubscriptions = this.getAllPushSubscriptionModels();
+      return Object.values(pushSubscriptions).find(
+        (subscription) => subscription.data.token === pushToken,
```

Comment:
Why not continue using `getSubscriptionOfTypeWithToken()`?

Reply:
I did this because I don't know what `SubscriptionType` the pushSubscription is. I could put a random push `SubscriptionType` to pass into `getSubscriptionOfTypeWithToken()` or make the parameter optional instead. Do you have any thoughts?

Reply:
There are other spots in this file where we can make this same change to simplify the code. Can we update those as well?

Reply:
I found a spot here: https://github.com/OneSignal/OneSignal-Website-SDK/pull/1179/commits/a2a8ef1a48ff7260d52364786b25ab8576d347ec

Did I miss anything else?